Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,354 @@
+  have ⟨a, ha, hf2'⟩ := Metric.eventually_nhds_iff_ball.mp hf2
+  have hcont := (hf2' t₀ (Metric.mem_ball_self ha)).continuousAt
+  rw [continuousAt_def, hf1] at hcont
+  have hnhds : f ⁻¹' (interior (extChartAt I x₀).target) ∈ nhds t₀ :=
+    hcont _ (isOpen_interior.mem_nhds ((I.isInteriorPoint_iff).mp hx))
+  rw [← eventually_mem_nhds] at hnhds
+  -- obtain a neighbourhood `s` so that the above conditions both hold in `s`
+  obtain ⟨s, hs, haux⟩ := (hf2.and hnhds).exists_mem
+  -- prove that `γ := (extChartAt I x₀).symm ∘ f` is a desired integral curve
+  refine ⟨(extChartAt I x₀).symm ∘ f,
+    Eq.symm (by rw [Function.comp_apply, hf1, PartialEquiv.left_inv _ (mem_extChartAt_source ..)]),
+    s, hs, ?_⟩
+  intros t ht
+  -- collect useful terms in convenient forms
+  let xₜ : M := (extChartAt I x₀).symm (f t) -- `xₜ := γ t`
+  have h : HasDerivAt f (x := t) <| fderivWithin ℝ (extChartAt I x₀ ∘ (extChartAt I xₜ).symm)
+    (range I) (extChartAt I xₜ xₜ) (v xₜ) := (haux t ht).1
+  rw [← tangentCoordChange_def] at h
+  have hf3 := mem_preimage.mp <| mem_of_mem_nhds (haux t ht).2
+  have hf3' := mem_of_mem_of_subset hf3 interior_subset
+  have hft1 := mem_preimage.mp <|
+    mem_of_mem_of_subset hf3' (extChartAt I x₀).target_subset_preimage_source
+  have hft2 := mem_extChartAt_source I xₜ
+  -- express the derivative of the integral curve in the local chart
+  refine ⟨(continuousAt_extChartAt_symm'' _ _ hf3').comp h.continuousAt,
+    HasDerivWithinAt.hasFDerivWithinAt ?_⟩
+  simp only [mfld_simps, hasDerivWithinAt_univ]
+  show HasDerivAt ((extChartAt I xₜ ∘ (extChartAt I x₀).symm) ∘ f) (v xₜ) t
+  -- express `v (γ t)` as `D⁻¹ D (v (γ t))`, where `D` is a change of coordinates, so we can use
+  -- `HasFDerivAt.comp_hasDerivAt` on `h`
+  rw [← tangentCoordChange_self (I := I) (x := xₜ) (z := xₜ) (v := v xₜ) hft2,
+    ← tangentCoordChange_comp (x := x₀) ⟨⟨hft2, hft1⟩, hft2⟩]
+  apply HasFDerivAt.comp_hasDerivAt _ _ h
+  apply HasFDerivWithinAt.hasFDerivAt (s := range I) _ <|
+    mem_nhds_iff.mpr ⟨interior (extChartAt I x₀).target,
+      subset_trans interior_subset (extChartAt_target_subset_range ..),
+      isOpen_interior, hf3⟩
+  dsimp only [xₜ]
+  nth_rw 4 [← (extChartAt I x₀).right_inv hf3']
+  exact hasFDerivWithinAt_tangentCoordChange ⟨hft1, hft2⟩
+
+/-- For any continuously differentiable vector field defined on a manifold without boundary and any
+  chosen starting point `x₀ : M`, an integral curve `γ : ℝ → M` exists such that `γ t₀ = x₀` and the
+  tangent vector of `γ` at `t` coincides with the vector field at `γ t` for all `t` within an open
+  interval around `t₀`. -/
+lemma exists_isIntegralCurveAt_of_contMDiffAt_boundaryless [BoundarylessManifold I M]
+    (hv : ContMDiffAt I I.tangent 1 (fun x ↦ (⟨x, v x⟩ : TangentBundle I M)) x₀) :
+    ∃ (γ : ℝ → M), γ t₀ = x₀ ∧ IsIntegralCurveAt γ v t₀ :=
+  exists_isIntegralCurveAt_of_contMDiffAt t₀ hv (BoundarylessManifold.isInteriorPoint I)
+
+section Naturality
```

Comment:
Can you add a doc-string to this section, and - more importantly, mention *this* result in the high-level docstring?

Reply:
Yes of course! If we're going to introduce `f`-relatedness and so on, which will be in a different file, this section should probably also go in a different file.